test-e2e/test_calico_networking.py

```diff
@@ -0,0 +1,111 @@
+"""
+Tests for calico network containers Different from tests under
+packages/dcos-integration-test/test_calico_networking.py which tests only with
+default parameters, current tests also take into consideration scenerios when
+different configurations exposed to the operators are enabled.
+"""
+import uuid
+from pathlib import Path
+from typing import Iterator
+
+import pytest
+
+from _pytest.fixtures import SubRequest
+from cluster_helpers import wait_for_dcos_oss
+from dcos_e2e.backends import Docker
+from dcos_e2e.cluster import Cluster
+from dcos_e2e.node import Node, Output
+from passlib.hash import sha512_crypt
+
+
+superuser_username = str(uuid.uuid4())
+superuser_password = str(uuid.uuid4())
+
+
+def assert_system_unit_state(node: Node, unit_name: str, active: bool=True) -> None:
+    result = node.run(
+        args=["systemctl show {}".format(unit_name)],
+        output=Output.LOG_AND_CAPTURE,
+        shell=True,
+    )
+    unit_properties = result.stdout.strip().decode()
+    if active:
+        assert "ActiveState=active" in unit_properties
+    else:
+        assert "ActiveState=inactive" in unit_properties
+        assert "ConditionResult=no" in unit_properties
+
+
+@pytest.fixture(scope="module")
+def calico_ipip_cluster(docker_backend: Docker, artifact_path: Path,
```

Comment:
```suggestion
def calico_ipip_cluster(e2e_backend: Docker, artifact_path: Path,
```

Reply:
Other backends are not recommended as the input. So let's focus on docker_backend

Reply:
Sorry, I misunderstood here.
But we don't have an `e2e_backend` fixture.

Reply:
Ah, did not notice that this is a fixture, thought that this is just a param. Sorry for the confusion!